### ISSUE TITLE
Training iteration time is too long

### ISSUE THREAD

**Issue by MaxMorning** [kai422/IART#7]
Hi!
I want to train my own gray-scale video dataset with this model. I just modified the number of in channels and out channels, and used my own dataloader, but I found it's pretty slow in each iteration. I used the official code to train and enabled AMP during training with 6 frames as input.
It would cost about 8 seconds per iter with batch size 4 on single Tesla V100, which is a thousand times more than the time given in the paper.
Here is a part of logs:
```
...
2024-05-08 16:38:01,914 INFO: [IART_..][epoch: 39, iter:   4,075, lr:(1.999e-04,2.499e-05,)] [eta: 30 days, 19:36:29, time (data): 8.109 (0.003)] l_pix: 2.5940e-02 
2024-05-08 16:38:09,855 INFO: [IART_..][epoch: 39, iter:   4,076, lr:(1.999e-04,2.499e-05,)] [eta: 30 days, 19:35:04, time (data): 8.106 (0.003)] l_pix: 2.6145e-02 
```

I also trained on single RTX 4070Ti with batch size 1. I found that the forward function took about 65 ms but the backward function took about 1800 ms, which seems to be abnormal and also doesn't match the time given in the paper.

I wonder if there are any details I haven't noticed that are causing these problems, thanks!

**Comment by kai422** [comment]
Hi

For 6 frames input, you can disable all gradient checkpointing by set https://github.com/kai422/IART/blob/4c8ae5968f4625af5b0eb43827fa962b022904b0/archs/iart_arch.py#L31C35-L31C60:
to  ` use_checkpoint = [False, False, False, False],`.

You can also adjust it according to your GPU memory size.
Thanks for pointing out of this!

Regards,
Kai

**Comment by DachunKai** [comment]
@kai422 Hi, kai. Could you explain the function of the parameter `use_checkpoint`? Why for 6 frames input, can we set `use_checkpoint` = [False, False, False, False]?

**Comment by PhuTran1005** [comment]
Hi,
If we change use_checkpoint = [False, False, False, False], it will not affect to the performance of the model, Am i right?
And in the paper, do u also use this setting?
Thanks

**Comment by kai422** [comment]
Hi, the option for using gradient checkpointing will not affect gradient computation, it only trade-off computation for memroy saving.  (https://pytorch.org/docs/stable/checkpoint.html)

Regards,
Kai

**Comment by PhuTran1005** [comment]
I see, thanks for quick reply
